packages/encrypt-upload-client/src/encrypt.js

```diff
@@ -0,0 +1,152 @@
+    
+                  /** @type {Type.EncryptedMetadataInput} */
+                const uploadData = {
+                  encryptedDataCID: root.cid.toString(),
+                  identityBoundCiphertext,
+                  plaintextKeyHash,
+                  accessControlConditions: /** @type {[Record<string, any>]} */ (
+                    /** @type {unknown} */ (accessControlConditions)
+                  )
+                }
+    
+                const encryptedMetadata = EncryptedMetadata.create(uploadData)
+                const { cid, bytes } = await encryptedMetadata.archiveBlock()
+                controller.enqueue({ cid, bytes })
+              }
+            }))
+            .pipeThrough(new CARWriterStream())
+        }
+    })
+}
+
+/**
+ * Encrypt a file
+ * @param {import('@lit-protocol/lit-node-client').LitNodeClient} litClient - The Lit client
+ * @param {Type.BlobLike} file - The file to encrypt
+ * @param {import('@lit-protocol/types').AccessControlConditions} accessControlConditions - The access control conditions
+ * @returns {Promise<Type.EncryptedPayload>} - The encrypted file
+ */
+const encryptFile = async (litClient, file, accessControlConditions) => {
+    const {cipher, dataToEncrypt} = createEncryptKey()
+
+    const { ciphertext, dataToEncryptHash } = await Lit.encryptString(
+        { dataToEncrypt, accessControlConditions },
+        litClient
+    )
+
+    /** @type {Type.BlobLike} */
+    const encryptedBlobLike = {
+        stream: () => {
+            const encryptStream = new TransformStream({
+                transform: async (chunk, controller) => {
+                    const encryptedChunk = cipher.update(chunk)
+                    if (encryptedChunk.length) {
+                        controller.enqueue(encryptedChunk)
+                    }
+                },
+                flush: (controller) => {
+                    const final = cipher.final()
+                    if (final.length) {
+                        controller.enqueue(final)
+                    }
+                }
+            })
+            return file.stream().pipeThrough(encryptStream)
+        }
+    }
+
+    return {
+        identityBoundCiphertext: ciphertext,
+        plaintextKeyHash: dataToEncryptHash,
+        encryptedBlobLike
+    }
+}
+
+const createEncryptKey = () => {
+    // Generate a random symmetric key and initialization vector
+  const symmetricKey = randomBytes(32) // 256 bits for AES-256
+  const initializationVector = randomBytes(16) // 16 bytes for AES
+
+  const cipher = createCipheriv(ENCRYPTION_ALGORITHM, symmetricKey, initializationVector)
+
+  // Combine key and initializationVector for Lit encryption
+  const dataToEncrypt = Buffer.concat([symmetricKey, initializationVector]).toString('base64')
+
+  /**
+   * TODO: use web crypto API. 
+   * ISSUE: web crypto does not support streaming encryption. We can hack it using the 'AES-CBC', but this also adds a security risk. We need to evaluate the risk.
```

Comment:
What is the issue here exactly - it looks like `dataToEncrypt` is only 48 bytes. Should be able to keep that in memory no?